File: snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/async.cs
using System;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        WriteTextAsync();
    }

    static async void WriteTextAsync()
    {
        // Set a variable to the Documents path.
        string docPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);

        // Write the specified text asynchronously to a new file named "WriteTextAsync.txt".
        using (StreamWriter outputFile = new StreamWriter(Path.Combine(docPath, "WriteTextAsync.txt")))
        {
            await outputFile.WriteAsync("This is a sentence.");
        }
    }
}

File: snippets/visualbasic/VS_Snippets_CLR/conceptual.basicio.textfiles/vb/async.vb
Imports System
Imports System.IO

Class WriteTextAsync

    Public Shared Async Sub Main()

        ' Set a variable to the Documents path.
        Dim docPath As String = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments)

        ' Write the text asynchronously to a new file named "WriteTextAsync.txt".
        Using outputFile As New StreamWriter(Path.Combine(docPath, Convert.ToString("WriteTextAsync.txt")))
            Await outputFile.WriteAsync("This is a sentence.")
        End Using
    End Sub

End Class
